src/shared/logger/sharedFileTransport.ts

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import TransportStream from 'winston-transport'
+import { FileSystemCommon } from '../../srcShared/fs'
+import * as vscode from 'vscode'
+import globals from '../extensionGlobals'
+import { MESSAGE } from './debugConsoleTransport'
+import { WinstonToolkitLogger } from './winstonToolkitLogger'
+
+interface LogEntry {
+    level: string
+    message: string
+    /** This is the formatted message from {@link WinstonToolkitLogger} in the winston.createLogger() call */
+    [MESSAGE]: string
+}
+
+export const flushIntervalMillis = 1000
+
+/**
+ * TODO: Figure out why the log file is not being properly appended to
+ *
+ * Problem:
+ * - Doing multiple log calls will not properly append to the log file
+ * - Only the last log call is written to the log file
+ * - Looking in to the append() call, when it first reads the current content
+ *   it would not have any of the previous logs in it.
+ * - The `next()` method is intended to signal that the current log is done being
+ *   written to the file but that is not working by the looks of it.
+ * Temporary Solution:
+ * - I created a buffer that buffers incoming logs and batch writes them after
+ *   a certain timeout. This seems to improve things
+ * - Also the current append implementation is inefficient due to the underlying api
+ *   (reads whole file, concatenates, then writes) so buffering is beneficial here.
+ */
+export class SharedFileTransport extends TransportStream {
```

Comment:
Why not just `FileTransport`? I don't think we want to prefix every class with "shared"/browser/node?

Reply:
There is a File transport built in to winston already, so I wanted to put "shared" in this case to differentiate it